apps/molgenis-components/src/components/account/MolgenisSession.vue

```diff
@@ -143,6 +168,17 @@ export default defineComponent({
         this.loadSettings(schemaSettings);
         this.session.manifest = schemaSettings._manifest;
       }
+      //set default locale
+      if (this.session.locale === undefined) {
+        //get from cookie
+        const lang = cookies.get("MOLGENIS.locale");
+        if (lang) {
+          this.session.locale = lang;
+        }
+      }
+      //get the map
+      console.log(this.settingsMap);
+      this.session.settingsMap = this.settingsMap;
```

Comment:
double bookkeeping, maybe just have session.settingsMap? Alternatively to not have redundancy on the session object, use the local map and also create a new map in places where the session is also used.

Reply:
then we are back to the start. In practice then in application state the session.settings = Record and not a ISetting[].
While when we get it from the backend it is actually an ISetting.